src/middleware/middleware-debug.ts

```diff
@@ -0,0 +1,26 @@
+import { Middleware } from "retes";
+
+import { createDebug } from "../debug";
+
+export const createMiddlewareDebug = (middleware: string) =>
+  createDebug(`Middleware:${middleware}`);
+
+type DebugFactory = (handlerName: string) => (msg: string, ...args: any[]) => void;
+
+/**
+ * Experimental. Needs to be tested and evaluated on security
+ */
+export const withReqResDebugging =
+  (debugFactory: DebugFactory = createMiddlewareDebug): Middleware =>
+  (handler) =>
+  async (request) => {
+    const debug = debugFactory(handler.name);
+
+    debug("Called with request %j", request);
+
+    const response = await handler(request);
+
+    debug("Responded with response %j", response);
+
+    return response;
+  };
```

Comment:
@krzysztofwolski I added this one, but I'd like not to connect it yet. Maybe it should be something that client is connecting explicitly? I'm worried that logging everything may leak data

Reply:
Agree, that's a helper which I would like to use occasionally